consensus/core/src/network/tonic_network.rs

```diff
@@ -627,6 +634,10 @@ impl<S: NetworkService> NetworkManager<S> for TonicManager {
                             continue;
                         }
                     };
+
+                    tcp_connection_metrics.socket_send_buffer_size.set(socket.send_buffer_size().unwrap_or(0) as i64);
+                    tcp_connection_metrics.socket_recv_buffer_size.set(socket.recv_buffer_size().unwrap_or(0) as i64);
+
```

Comment:
Do you not need to do an explicit close for the ephemeral port after you check the buffers?

Reply:
The doc says `The underlying socket is closed when the TcpSocket value is dropped.`.